idunn/api/urlsolver.py

```diff
@@ -0,0 +1,42 @@
+import hmac
+from secrets import token_bytes
+from urllib.parse import quote
+
+import httpx
+from fastapi import HTTPException
+from fastapi.responses import RedirectResponse
+from idunn import settings
+
+
+client = httpx.AsyncClient()
+base_url = settings.get("BASE_URL")
+secret = settings.get("SECRET").encode()
+
+
+def resolve_url(url: str) -> str:
+    """
+    Idunn's URL that can be provided to redirect to the same page as the input
+    URL would.
+    """
+    return base_url + f"v1/redirect?url={quote(url)}&hash={hash_url(url)}"
```

Comment:
`quote` does not encode `/`, because it's "is intended for quoting the path section of a URL" by default. 
I think that it would be safer to quote them here.
```suggestion
    return base_url + f"v1/redirect?url={quote(url, safe='')}&hash={hash_url(url)}"
```

Reply:
Yep, it doesn't cost anything to be careful here